packages/postgres-database/src/simcore_postgres_database/utils_groups_extra_properties.py

```diff
@@ -0,0 +1,150 @@
+    )
+    assert result  # nosec
+
+    rows: list[RowProxy] | None = await result.fetchall()
+    assert rows is not None  # nosec
+    return rows
+
+
+def _merge_extra_properties_booleans(
+    instance1: GroupExtraProperties, instance2: GroupExtraProperties
+) -> GroupExtraProperties:
+    merged_properties: dict[str, Any] = {}
+    for field in fields(instance1):
+        value1 = getattr(instance1, field.name)
+        value2 = getattr(instance2, field.name)
+
+        if isinstance(value1, bool):
+            merged_properties[field.name] = value1 or value2
+        else:
+            merged_properties[field.name] = value1
+    return GroupExtraProperties(**merged_properties)  # pylint: disable=missing-kwoa
+
+
+@dataclass(frozen=True, slots=True, kw_only=True)
+class GroupExtraPropertiesRepo:
+    @staticmethod
+    async def get(
+        connection: SAConnection, *, gid: int, product_name: str
+    ) -> GroupExtraProperties:
+        get_stmt = sqlalchemy.select(groups_extra_properties).where(
+            (groups_extra_properties.c.group_id == gid)
+            & (groups_extra_properties.c.product_name == product_name)
+        )
+        result = await connection.execute(get_stmt)
+        assert result  # nosec
+        if row := await result.first():
+            return GroupExtraProperties.from_row(row)
+        raise GroupExtraPropertiesNotFound(f"Properties for group {gid} not found")
+
+    @staticmethod
+    async def get_aggregated_properties_for_user(
+        connection: SAConnection,
+        *,
+        user_id: int,
+        product_name: str,
+    ) -> GroupExtraProperties:
+        rows = await _list_table_entries_ordered_by_group_type(
+            connection, user_id, product_name
+        )
+        merged_standard_extra_properties = None
+        for row in rows:
+            group_extra_properties = GroupExtraProperties.from_row(row)
+            match row.type:
+                case GroupType.PRIMARY:
+                    # this always has highest priority
+                    return group_extra_properties
+                case GroupType.STANDARD:
+                    if merged_standard_extra_properties:
+                        merged_standard_extra_properties = (
+                            _merge_extra_properties_booleans(
+                                merged_standard_extra_properties,
+                                group_extra_properties,
+                            )
+                        )
+                    else:
+                        merged_standard_extra_properties = group_extra_properties
+                case GroupType.EVERYONE:
+                    # if there are standard properties, they take precedence
+                    return (
+                        merged_standard_extra_properties
+                        if merged_standard_extra_properties
+                        else group_extra_properties
+                    )
```

Comment:
Maybe raise a warning/error when there is not match (the situation when you have "literal 4" from the previous SQL query)

Reply:
not sure I get your point, it actually raise here a GroupExtraPropertiesNotFound when there is no match.

Reply:
aa you are right :+1: